packages/renderers/renderer-lit/server.js

```diff
@@ -28,8 +28,13 @@ function* render(tagName, attrs, children) {
   const instance = new LitElementRenderer(tagName);
 
   // LitElementRenderer creates a new element instance, so copy over.
+  const Ctr = getCustomElementConstructor(tagName);
   for (let [name, value] of Object.entries(attrs)) {
-    instance.setAttribute(name, value);
+    if (name in Ctr.prototype) {
```

Comment:
In our React wrapper, which does something similar, we only set properties for names in the prototype but _not_ in `HTMLElement.prototype`. This let's React do any special handling of properties that it does for regular HTML elements. I'm not sure if Astro has custom logic we can defer to, but if so it'd be good to do something like this: https://github.com/lit/lit/blob/main/packages/labs/react/src/create-component.ts#L151

Also, the React wrapper computes the properties once and reuses them to be faster than a prototype lookup every time. Is any state persisted between renders/pages that could store that?

cc @matthewp

Reply:
Note that `HTMLElement` is the one from Lit's shim and doesn't contain much, but yeah filtering out those things makes sense to me.